src/lib/TopLevel.hs

```diff
@@ -731,13 +731,7 @@ checkPass name cont = do
     return result
 #ifdef DEX_DEBUG
   logTop $ MiscLog $ "Running checks"
-  let allowedEffs = case name of
-                      LowerPass    -> OneEffect InitEffect
-                      LowerOptPass -> OneEffect InitEffect
-                      VectPass     -> OneEffect InitEffect
-                      _            -> mempty
-  {-# SCC afterPassTypecheck #-} (liftExcept =<<) $ liftEnvReaderT $
-    withAllowedEffects @CoreIR allowedEffs $ checkTypesM result
+  checkTypesM result
```

Comment:
Where do we now check that top-level forms do not have impermissible effects?

Reply:
We don't. Now that we have to pass the effects around explicitly, it's a bit more awkward. But I can add some checks back.